docs/man/rpm-common.8.scd

```diff
@@ -0,0 +1,229 @@
+RPM-COMMON(8)
+
+# NAME
+rpm-common - operations and options common to the main *rpm*(8) executables
+
+# SYNOPSIS
+_RPMCMD_ [options] {*-?*|*--help*}
+
+_RPMCMD_ [options] *--showrc*
+
+_RPMCMD_ [options] *--version*
+
+# DESCRIPTION
+The following *rpm* executables share common configuration and the
+options and operations documented in this manual:
+- *rpm*(8)
+- *rpmdb*(8)
+- *rpmkeys*(8)
+- *rpmbuild*(1)
+- *rpmsign*(1)
+- *rpmspec*(1)
+
+# OPERATIONS
+*-?*,
+*--help*
+	Print a longer usage message than normal.
+
+*--version*
+	Print a single line containing the version number of *rpm* being
+	used.
+
+*--showrc*
+	Dump information about *rpm* configuration to standard output.
+
+# ARGUMENTS
+The common operations take no arguments.
+
+# OPTIONS
+*--color* <*never*|*auto*|*always*>
+	Use terminal colors for highlighting error and debug message.
+	Default is turning colors on for ttys only (*auto*).
+
+*--dbpath* _DIRECTORY_
+	Use the database in _DIRECTORY_ rather than the default path
+	_/var/lib/rpm_.
+
+*-D* '_MACRO_ _EXPR_',
+*--define*='_MACRO_ _EXPR_'
+	Defines _MACRO_ with value _EXPR_.
+
+*-E* '_EXPR_',
+*--eval*='_EXPR_'
+	Prints macro expansion of _EXPR_.
+
+*--load* _FILE_
+	Load an individual macro file.
+
+*--macros* _FILELIST_
+	Replace the list of macro files to be loaded with _FILELIST_.
+	See *Macro Configuration* for details.
+
+*--nodigest*
+	Don't verify package or header digests when reading.
+
+*--nohdrchk*
+	Don't verify database header(s) when retrieved.
+
+*--nosignature*
+	Don't verify package or header signatures when reading.
+
+*--pipe* _CMD_
+	Pipes the output of *rpm* to the command _CMD_.
+
+*--predefine*='_MACRO_ _EXPR_'
+	Defines _MACRO_ with value _EXPR_ before loading macro files.
+	This is a very rare special-purpose switch, use *--define*
+	for the common macro definition needs.
+
+*--quiet*
+	Print as little as possible - normally only error messages will be
+	displayed.
+
+*--rcfile* _FILELIST_
+	Replace the default list of configuration files to be read with
+	_FILELIST_.  See *rpmrc Configuration* for details.
+
+*--root* _DIRECTORY_
+	Use the file system tree rooted at _DIRECTORY_ for all operations.
+	Note that this means the database within _DIRECTORY_ will be used
+	for dependency checks and any scriptlet(s) (e.g. *%post* if
+	installing, or *%prep* if building, a package) will be run after a
+	chroot(2) to _DIRECTORY_.
+
+	Note that rpm assumes the environment inside the root is set up by
+	the caller, such as any mounts needed for the operation inside the
+	root directory.
+
+*--undefine*='_MACRO_'
+	Undefines _MACRO_.
+
+*-v*,
+*--verbose*
+	Print verbose information - normally routine progress messages will
+	be displayed. Supply more than once to increase verbosity, eg. *-vv*.
+
+# DEBUG OPTIONS
```

Comment:
"Debug" is not an operation, though. Should we still separate these options like that?

Reply:
Hmm, there seem to be duplicate debug options now, oops :sweat_smile:

Reply:
Understood, I haven't yet gone through the whole PR and indeed, that's what the updated rpm.8 page has. And I agree of course, there doesn't have to a be strict rule for "only separate by operation" as it can grow very long, the option list, otherwise.

Reply:
OTOH, it's not explicitly discouraged in the template either, it's just not mentioned... so meh, don't bother :smile: